drivers/platform/xilinx/irq.c

```diff
@@ -279,6 +279,45 @@ int32_t irq_register_callback(struct irq_ctrl_desc *desc, uint32_t irq_id,
 	return FAILURE;
 }
 
+/**
+ * @brief Set interrupt trigger level.
+ * @param desc - The IRQ controller descriptor.
+ * @param irq_id - Interrupt identifier.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t irq_trigger_level_set(struct irq_ctrl_desc *desc, uint32_t irq_id,
+		enum irq_trig_level trig)
+{
+	struct xil_irq_desc *xil_dev = desc->extra;
+
+	switch(xil_dev->type) {
+	case IRQ_PS:
+#ifdef XSCUGIC_H
+		;
+		uint8_t priority, trigger;
```

Comment:
Isn't better if you put this at the beggining of the function and use UNUSED_PARAM(priority) in the else?

Reply:
Did not apply yet. util.h not included in the irq yet. Do we want to make that inclusion?